### ISSUE TITLE
move test code that connects to a db into separate module

### ISSUE THREAD

**Issue by SingingBush** [mysql-d/mysql-native#233]
I've started a branch ([SingingBush : pipeline/integration_testing](https://github.com/SingingBush/mysql-native/tree/pipeline/integration_testing)) to move some of the testing out of the main source. 

The code I want to do away with is

 - source/app.d
 - source/mysql/test/*.d

Those source files are all running tests against a MySQL instance, they pulling into a separate test project. I also see the existing examples directory as being pointless. it's used as a test and it's not particularly good as an example anyway.

In my branch I've got a tests/ directory as so:

```
tests/
    source/app.d   <-- covers same stuff that `source/app.d` and `examples/homePage/example.d` used to have
    dub.sdl        <-- has x2 configurations, one of which uses vibe 
    docker-compose.yml   <-- allows user to locally spin up MySQL in same way that the CI will
    README.md
``` 

It will need some more work to get all the test scenarios from `source/mysql/test/regression.d` moved across. I'll raise a merge request.

**Comment by schveiguy** [comment]
eek, you move too fast! I'm already working on this and nearly done.

I've moved all test code out of the main library, and moved app.d to its own thing (it's really not a test).

Just take a break for a bit, after I get my branch going, maybe you can help me set up the CI for it.

**Comment by SingingBush** [comment]
ok, no problem. I'll wait for your changes

**Comment by schveiguy** [comment]
See my branch. I think it's good (very easy to update the testing, great job!): https://github.com/schveiguy/mysql-native/tree/splittests

Look at that beautiful straightforward dub.sdl file ;) If it passes tests, I'll make a PR, and we can discuss.

**Comment by SingingBush** [comment]
closing as this is resolved by #235
